test/cli/test_cli_functions.sh

```diff
@@ -180,6 +180,34 @@ function noobaa_install {
     kuberun describe noobaa
 }
 
+function check_core_config_map {
+    kuberun get configmap noobaa-config
+    check_change_debug_level_in_config_map
+}
+
+function check_change_debug_level_in_config_map {
+    local patch='{"data":{"NOOBAA_LOG_LEVEL":"2"}}'
+    local cm_debug_level="2"
+    local core_debug_level=$(kuberun silence exec noobaa-core-0 -- printenv NOOBAA_LOG_LEVEL)
+
+    kuberun silence patch configmap noobaa-config -p ${patch}
+
+    while [[ "${core_debug_level}" != "${cm_debug_level}" ]]
+    do
+        echo_time "💬  Waiting for NOOBAA_LOG_LEVEL core env var to match the noobaa-config"
+        sleep 10
+        core_debug_level=$(kuberun silence exec noobaa-core-0 -- printenv NOOBAA_LOG_LEVEL)
+    done < <(sleep 360)
```

Comment:
Why do we want to sleep that long? 
+ I think it is more straightforward to do the sleep inside the loop:
```suggestion
    while [[ "${core_debug_level}" != "${cm_debug_level}" ]]
    do
        echo_time "💬  Waiting for NOOBAA_LOG_LEVEL core env var to match the noobaa-config"
        sleep 10
        core_debug_level=$(kuberun silence exec noobaa-core-0 -- printenv NOOBAA_LOG_LEVEL)
        sleep 360
    done 
```

Reply:
the `sleep 360` outside the loop is just a timeout so we won't get stuck waiting for the update, the way I see it in your suggestion we are checking one time if the update occurred and than sleeping 360 seconds.

Reply:
I don't see how it act as a timeout:
```
$ count=1 ; while true ; do echo ${count} ; ((count++)) ; sleep 2 ; done < <(sleep 6)
1
2
3
4
5
6
7
8
9
10
11
12
```
So if you are missing some sleep inside the loop, you just keep checking, In your case you are probably succeeding, have the loop ends.

Reply:
In your example: `done < <(sleep 360)` you are passing the `fd` as an input to the loop, but you are not doing anything with it

Reply:
you are right, fixed that with a timeout condition in the while loop.